### PR TITLE
Fix heading margins per #1668

### PR DESCRIPTION
**Pull request by dgennetten**

### CODE REVIEW

File: src/scss/grommet-core/_objects.heading.scss [hunk 1]
```diff
@@ -53,3 +53,9 @@ h6.#{$grommet-namespace}heading {
   text-transform: uppercase;
   letter-spacing: 0.2em;
 }
+
+.#{$grommet-namespace}heading--margin-large,
+.#{$grommet-namespace}heading--margin-medium,
+.#{$grommet-namespace}heading--margin-small {
+  margin-top: 0px;
```
Comment (alansouzati) [inline src/scss/grommet-core/_objects.heading.scss]: why is that needed?

**Comment by dgennetten** [comment]
Currently, margin prop on Heading incorrectly sets both top and bottom
margin. Should only set bottom. #1668

—Douglas

**Comment by alansouzati** [comment]
do we know why it sets margin-top in the first place? can we avoid that to happen?

**Comment by dgennetten** [comment]
I believe it is coming from the mixin _tools.margin.scss and applied symmetrically. We cannot isolate heading margins there.

**Comment by dgennetten** [comment]
@alansouzati, I think this is good to go as is.

**Comment by alansouzati** [comment]
I would defer that to @ericsoderberghp. I believe this will affect the layout of existing 1.x applications and can be classified as a breaking change.
